deckhandler.c

```diff
@@ -79,45 +79,27 @@ deck_init_dh (st_deck_dh *deck_dh)
 void
 deck_shuffle_dh (st_deck_dh *deck_dh)
 {
-  int j = 0;
-  bool status[CARDS_IN_DECK];
-
-  do
-  {
-    status[j++] = 1;
-  }while (j < CARDS_IN_DECK);
-
-  j = 0;
-
-  int card = 0;
-
-  st_deck_dh temp_deck;
-
-  while (j < CARDS_IN_DECK)
+  int x;
+  for (x = 0; x < 3; x++)
   {
-    /* card will a random number between 0 and 51 */
-    card = rand () % CARDS_IN_DECK;
-
-    if (status[card])
+    int i = 0;
+    while (i < CARDS_IN_DECK)
     {
-      temp_deck.card[j].face_val = deck_dh->card[card].face_val;
-      temp_deck.card[j].suit = deck_dh->card[card].suit;
-
-      /* switch status[card] to 0 so it won't get used again */
-      status[card] = 0;
-      j++;
+      /* card will be a random number between 0 and 51 */
+      int rnd = rand () % CARDS_IN_DECK;
```

Comment:
Can't remember if rand can ever equal 1. If yes, doesn't that mean rnd could equal 52?
```suggestion
      int rnd = rand () % CARDS_IN_DECK;
      rnd = rnd == 52 ? 51 : rnd; // possible fix if this is a problem.
```

Reply:
Here's the output when I try as-is:

```
rnd = 28
rnd = 50
rnd = 45
rnd = 8
rnd = 27
rnd = 34
rnd = 24
rnd = 12
rnd = 2
rnd = 4
rnd = 46
rnd = 6
rnd = 28
rnd = 32
rnd = 7
rnd = 3
rnd = 18
rnd = 14
rnd = 2
rnd = 28
rnd = 50
rnd = 45
rnd = 12
rnd = 15
rnd = 44
rnd = 20
rnd = 39
rnd = 36
rnd = 3
rnd = 48
rnd = 20
rnd = 31
rnd = 46
rnd = 14
rnd = 15
rnd = 21
rnd = 48
rnd = 16
rnd = 34
rnd = 26
rnd = 20
rnd = 4
rnd = 33
rnd = 24
rnd = 13
rnd = 16
rnd = 27
rnd = 31
rnd = 6
rnd = 6
rnd = 35
rnd = 5
rnd = 51
rnd = 48
rnd = 48
rnd = 20
rnd = 44
rnd = 11
rnd = 4
rnd = 47
rnd = 7
rnd = 25
rnd = 3
rnd = 29
rnd = 15
rnd = 46
rnd = 51
rnd = 11
rnd = 10
rnd = 9
rnd = 14
rnd = 6
rnd = 41
rnd = 47
rnd = 6
rnd = 2
rnd = 39
rnd = 10
rnd = 9
rnd = 45
rnd = 16
rnd = 21
rnd = 26
rnd = 15
rnd = 17
rnd = 50
rnd = 35
rnd = 37
rnd = 10
rnd = 16
rnd = 9
rnd = 45
rnd = 17
rnd = 40
rnd = 51
rnd = 32
rnd = 34
rnd = 26
rnd = 19
rnd = 21
rnd = 11
rnd = 33
rnd = 3
rnd = 0
rnd = 4
rnd = 10
rnd = 31
rnd = 19
rnd = 20
rnd = 40
rnd = 13
rnd = 36
rnd = 37
rnd = 39
rnd = 51
rnd = 30
rnd = 38
rnd = 11
rnd = 16
rnd = 24
rnd = 27
rnd = 1
rnd = 45
rnd = 44
rnd = 41
rnd = 44
rnd = 0
rnd = 51
rnd = 18
rnd = 47
rnd = 48
rnd = 29
rnd = 29
rnd = 0
rnd = 6
rnd = 9
rnd = 10
rnd = 37
rnd = 29
rnd = 30
rnd = 1
rnd = 18
rnd = 42
rnd = 39
rnd = 5
rnd = 17
rnd = 17
rnd = 19
rnd = 28
rnd = 9
rnd = 19
rnd = 31
rnd = 10
rnd = 13
rnd = 51
rnd = 27
rnd = 33
rnd = 27
rnd = 3
rnd = 0
rnd = 23
rnd = 51
rnd = 5
rnd = 28
rnd = 27
rnd = 11
rnd = 37
rnd = 37
rnd = 24
rnd = 42
rnd = 43
rnd = 26
rnd = 8
rnd = 33
rnd = 41
rnd = 42
rnd = 27
rnd = 6
rnd = 9
rnd = 31
rnd = 44
rnd = 5
rnd = 39
rnd = 30
rnd = 18
rnd = 38
rnd = 34
rnd = 27
rnd = 14
rnd = 37
rnd = 3
rnd = 37
rnd = 12
rnd = 9
rnd = 13
rnd = 40
rnd = 48
rnd = 26
rnd = 25
rnd = 21
rnd = 17
rnd = 45
rnd = 23
rnd = 1
rnd = 2
rnd = 12
rnd = 43
rnd = 29
rnd = 46
rnd = 29
rnd = 37
rnd = 14
rnd = 34
rnd = 24
rnd = 45
rnd = 28
rnd = 10
rnd = 27
rnd = 3
rnd = 0
rnd = 40
rnd = 7
rnd = 13
rnd = 0
rnd = 44
rnd = 2
rnd = 16
rnd = 40
rnd = 29
rnd = 18
rnd = 37
rnd = 22
rnd = 11
rnd = 36
rnd = 23
rnd = 13
rnd = 48
rnd = 15
rnd = 19
rnd = 19
rnd = 44
rnd = 4
rnd = 33
rnd = 26
rnd = 28
rnd = 2
rnd = 2
rnd = 14
rnd = 29
rnd = 33
rnd = 15
rnd = 45
rnd = 40
rnd = 4
rnd = 46
rnd = 32
rnd = 7
rnd = 38
rnd = 49
rnd = 36
rnd = 4
rnd = 10
rnd = 6
rnd = 15
rnd = 47
rnd = 5
rnd = 5
rnd = 19
rnd = 20
rnd = 24
rnd = 38
rnd = 40
rnd = 4
rnd = 48
rnd = 42
rnd = 8
rnd = 50
rnd = 20
rnd = 22
rnd = 4
rnd = 2
rnd = 13
rnd = 25
rnd = 42
rnd = 18
rnd = 47
rnd = 51
rnd = 25
rnd = 34
rnd = 24
rnd = 37
rnd = 38
rnd = 34
rnd = 43
rnd = 2
rnd = 29
rnd = 24
rnd = 35
rnd = 49
rnd = 45
rnd = 35
rnd = 11
rnd = 9
rnd = 39
rnd = 35
rnd = 0
rnd = 47
rnd = 10
rnd = 20
rnd = 45
rnd = 42
rnd = 22
rnd = 7
rnd = 15
rnd = 41
rnd = 25
rnd = 11
rnd = 16
rnd = 26
rnd = 45
rnd = 40
rnd = 11
rnd = 31
rnd = 22
rnd = 30
rnd = 9
rnd = 28
rnd = 2
rnd = 44
rnd = 1
rnd = 23
rnd = 27
rnd = 12
rnd = 33
rnd = 42
rnd = 48
rnd = 33
rnd = 37
rnd = 34
rnd = 29
rnd = 31
rnd = 24
rnd = 28
rnd = 14
rnd = 39
rnd = 45
rnd = 15
rnd = 50
rnd = 9
rnd = 41
rnd = 43
rnd = 25
rnd = 28
rnd = 51
rnd = 47
rnd = 6
rnd = 8
rnd = 51
rnd = 8
rnd = 29
rnd = 0
rnd = 32
rnd = 32
rnd = 13
rnd = 41
rnd = 23
rnd = 37
rnd = 22
rnd = 36
rnd = 19
rnd = 27
rnd = 43
rnd = 19
rnd = 3
rnd = 5
rnd = 6
rnd = 24
rnd = 48
rnd = 33
rnd = 33
rnd = 37
rnd = 0
rnd = 6
rnd = 13
rnd = 51
rnd = 30
rnd = 47
rnd = 36
rnd = 29
rnd = 4
rnd = 13
rnd = 30
rnd = 12
rnd = 21
rnd = 19
rnd = 1
rnd = 44
rnd = 32
rnd = 51
rnd = 29
rnd = 27
rnd = 26
rnd = 20
rnd = 46
rnd = 6
rnd = 2
rnd = 0
rnd = 30
rnd = 50
rnd = 9
rnd = 40
rnd = 12
rnd = 10
rnd = 46
rnd = 1
rnd = 37
rnd = 0
rnd = 49
rnd = 21
rnd = 6
rnd = 1
rnd = 10
rnd = 12
rnd = 41
rnd = 32
rnd = 7
rnd = 18
rnd = 24
rnd = 39
rnd = 17
rnd = 29
rnd = 42
rnd = 19
rnd = 26
rnd = 36
rnd = 1
rnd = 28
rnd = 12
rnd = 32
rnd = 2
rnd = 22
rnd = 20
rnd = 14
rnd = 8
rnd = 42
rnd = 16
rnd = 45
rnd = 19
rnd = 41
rnd = 15
rnd = 25
rnd = 18
rnd = 25
rnd = 37
rnd = 7
rnd = 33
rnd = 20
rnd = 25
rnd = 6
rnd = 7
rnd = 18
rnd = 11
rnd = 49
rnd = 37
rnd = 37
rnd = 9
rnd = 39
rnd = 41
rnd = 21
rnd = 47
rnd = 20
rnd = 19
rnd = 43
rnd = 34
rnd = 27
rnd = 33
rnd = 26
rnd = 21
rnd = 28
rnd = 15
rnd = 36
rnd = 1
rnd = 33
rnd = 37
rnd = 14
rnd = 16
rnd = 19
rnd = 34
rnd = 17
rnd = 1
rnd = 17
rnd = 35
rnd = 40
rnd = 42
rnd = 21
rnd = 2
rnd = 51
rnd = 36
rnd = 43
rnd = 49
rnd = 31
rnd = 11
rnd = 16
rnd = 22
rnd = 22
rnd = 44
rnd = 31
rnd = 24
rnd = 13
rnd = 8
rnd = 16
rnd = 25
rnd = 37
rnd = 25
rnd = 38
rnd = 28
rnd = 42
rnd = 33
rnd = 38
rnd = 7
rnd = 10
rnd = 4
rnd = 19
rnd = 51
rnd = 22
rnd = 40
rnd = 1
rnd = 22
rnd = 0
rnd = 20
rnd = 19
rnd = 31
rnd = 8
rnd = 11
rnd = 1
rnd = 6
rnd = 3
rnd = 8
rnd = 30
rnd = 44
rnd = 16
rnd = 22
rnd = 45
rnd = 30
rnd = 48
rnd = 8
rnd = 6
rnd = 14
rnd = 41
rnd = 44
rnd = 49
rnd = 0
rnd = 24
rnd = 16
rnd = 27
rnd = 47
rnd = 32
rnd = 4
rnd = 45
rnd = 32
rnd = 24
rnd = 12
rnd = 39
rnd = 32
rnd = 23
rnd = 16
rnd = 14
rnd = 3
rnd = 25
rnd = 45
rnd = 23
rnd = 17
rnd = 43
rnd = 17
rnd = 47
rnd = 39
rnd = 25
rnd = 1
rnd = 29
rnd = 42
rnd = 22
rnd = 3
rnd = 18
rnd = 46
rnd = 19
rnd = 21
rnd = 17
rnd = 28
rnd = 25
rnd = 10
rnd = 8
rnd = 26
rnd = 50
rnd = 24
rnd = 34
rnd = 50
rnd = 40
rnd = 49
rnd = 1
rnd = 13
rnd = 18
rnd = 24
rnd = 31
rnd = 9
rnd = 17
rnd = 2
rnd = 49
rnd = 18
rnd = 32
rnd = 2
rnd = 37
rnd = 2
rnd = 5
rnd = 3
rnd = 24
rnd = 1
rnd = 25
rnd = 18
rnd = 29
rnd = 26
rnd = 28
rnd = 13
rnd = 0
rnd = 27
```

Reply:
Your code is fine. I completely misunderstood it.

Reply:
Thanks for reviewing @Jammyjamjamman  :)